libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -99,6 +103,73 @@ string ABIFunctions::tupleEncoder(
 	});
 }
 
+string ABIFunctions::tupleDecoder(TypePointers const& _types, bool _fromMemory)
+{
+	string functionName = string("abi_decode_tuple_");
+	for (auto const& t: _types)
+		functionName += t->identifier();
+	if (_fromMemory)
+		functionName += "_fromMemory";
+
+	return createFunction(functionName, [&]() {
+		solAssert(!_types.empty(), "");
```

Comment:
Hm, actually doesn't this assert should be in the outer function?

Reply:
It does not really matter. The lambda function is guaranteed to be either executed immediately (otherwise, we could not take closure data by reference) or not at all (if the function is already defined).